plugins/source/aws/resources/services/cloudwatchlogs/data_protection_policies.go

```diff
@@ -34,8 +34,13 @@ func dataProtectionPolicy() *schema.Table {
 	}
 }
 func fetchDataProtectionPolicy(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	lg := parent.Item.(types.LogGroup)
+	if lg.DataProtectionStatus == "" { // Inactive Data Protection policy, don't attempt to fetch
```

Comment:
`DataProtectionStatus` is empty for groups without data protection set, and also the Get query comes up empty, which breaks the Primary Key.